[Bug]: master cannot compile after annoy removed from knowhere

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

failed to compile

### Expected Behavior

Compilation shall succeed

### Steps To Reproduce

```markdown
In file included from /go/src/github.com/milvus-io/milvus/internal/core/src/common/RangeSearchHelper.cpp:17:
/go/src/github.com/milvus-io/milvus/internal/core/src/common/Utils.h: In function 'std::string milvus::MatchKnowhereError(knowhere::Status)':
/go/src/github.com/milvus-io/milvus/internal/core/src/common/Utils.h:163:32: error: 'annoy_inner_error' is not a member of 'knowhere::Status'; did you mean 'hnsw_inner_error'?
  163 |         case knowhere::Status::annoy_inner_error:
      |                                ^~~~~~~~~~~~~~~~~
      |                                hnsw_inner_error
make[3]: *** [src/common/CMakeFiles/milvus_common.dir/build.make:160: src/common/CMakeFiles/milvus_common.dir/RangeSearchHelper.cpp.o] Error 1
make[3]: Leaving directory '/go/src/github.com/milvus-io/milvus/cmake_build'
make[2]: *** [CMakeFiles/Makefile2:1778: src/common/CMakeFiles/milvus_common.dir/all] Error 2
make[2]: Leaving directory '/go/src/github.com/milvus-io/milvus/cmake_build'
make[1]: *** [Makefile:146: all] Error 2
make[1]: Leaving directory '/go/src/github.com/milvus-io/milvus/cmake_build'
make: *** [Makefile:185: build-cpp-with-coverage] Error 1
```
```


### Milvus Log

_No response_

### Anything else?

_No response_

--- a/internal/core/src/common/Utils.h
+++ b/internal/core/src/common/Utils.h
@@ -160,8 +160,6 @@ MatchKnowhereError(knowhere::Status status) {
             return "err: index already trained";
         case knowhere::Status::faiss_inner_error:
             return "err: faiss inner error";
-        case knowhere::Status::annoy_inner_error:
-            return "err: annoy inner error";
         case knowhere::Status::hnsw_inner_error:
             return "err: hnsw inner error";
         case knowhere::Status::malloc_error:

